src/page/mod.rs

```diff
@@ -64,11 +64,15 @@ pub(crate) type Iter<'a, T, C> =
     std::iter::FilterMap<std::slice::Iter<'a, Slot<T, C>>, fn(&'a Slot<T, C>) -> Option<&'a T>>;
 
 pub(crate) struct Local {
+    // index of the first slot on the local free list
     head: CausalCell<usize>,
 }
 
 pub(crate) struct Shared<T, C> {
     remote: stack::TransferStack<C>,
+    // tracks the size of the local free_list by keeping the index of the current position of the
+    // start of the local free list. If local.head() > size, it means that the local free_list if
+    // full.
```

Comment:
I added these as notes for myself and accidentally got committed in. I'll remove them in the next commit.

Reply:
please feel free to add comments on anything that wasn't obvious — i'm totally fine with merging them to master, especially if we clean them up a bit! 

i do have a slight preference for merging them in a separate PR from the pool change, though, it keeps the git history a little clearer. but, if that's a lot of extra effort, i would be fine with merging new comments as part of this PR.

Reply:
I don't actually think this comment is entirely correct. This is the total size of this page, and it shouldn't ever change. The bit about determining if the free list is full is right, though. 

Maybe revise this to
```suggestion
    /// The total size of this page.
    ///
    /// If the head index of the local or remote free list is greater than the size of the page,
    /// then that free list is empty. If the head of both free lists is greater than `size`, then
    /// all slots on the page are in use.
```